tabbycat/standings/teams.py

```diff
@@ -256,16 +264,12 @@ def get_field(self):
             NullIf('debateteam__teamscore__votes_possible', 0, output_field=FloatField()) *
             self.adjs_per_debate)
 
-    def annotate(self, queryset, standings, round=None):
-        super().annotate(queryset, standings, round)
-
+    def get_annotation(self, round=None):
         # If the number of ballots carried by every team is an integer, then
         # it's probably (though not certainly) the case that there are no
         # "weird" cases causing any fractional numbers of votes due to
         # normalization. In that case, convert all metrics to integers.
-        if all(tsi.metrics[self.key] == int(tsi.metrics[self.key]) for tsi in standings.infoview()):
-            for tsi in standings.infoview():
-                tsi.metrics[self.key] = int(tsi.metrics[self.key])
+        return Cast(super().get_annotation(round), output_field=IntegerField())
```

Comment:
This changes the numerical value of the annotation in the situation alluded to in the comment just above it.

(To try it out, try a five-person panel, or just emulate one by changing the "votes possible" field in a team score in the database.)

Reply:
Oh now I see. Casting to integer won't require a modification to the queryset, just the standings, and the cast won't change any numerical values.

Reply:
The rounding can change things, because the metric isn't actually guaranteed to be an integer, at least not with the current implementation. It normalizes to a fixed number of adjudicators (currently 3), so that every debate has equal weight, even when panels are of different sizes, so if (e.g.) a team got 3 judges from a panel of 5, the value of the metric will be 1.8, but the cast will round it to 2.

Reply:
I'm talking about your approach, before this change. If the value was 1.8, no values would be rounded as 1.8 != 2.